### ISSUE TITLE
Adapt testcode as regtest-driver

### ISSUE THREAD

**Issue by dev-zero** [cp2k/cp2k#770]
[testcode](https://testcode.readthedocs.io) is a more generalized Python-based implementation of our regtest script. Wannier90 for example adopted it for their testing.

What they're lacking is support for farming (used for the many short tasks we have), but this could be implemented as an extension of `testcode` by either using a helper doing `MPI_Comm_spawn` (preferred if supported by popular implementations) or by wrapping tests inside generated CP2K farming inputs.

**Comment by oschuett** [comment]
I think we should continue to roll our own regtest script. 

I've started to [port](https://github.com/oschuett/cp2k/blob/regtest_py/tools/regtesting/do_regtest.py) the `do_regtest` script to Python, albeit with low priority. My main motivation is to replace the farming mode with a master-work scheme based on the new [RUN command](https://github.com/cp2k/cp2k/commit/8228f8cf641eed97787c2b3a5c51252b896f3e4b#diff-a7d4ae73627192da9d19578e3bd59624) which I added to the shell. This will speed up tests with high startup costs, like CUDA and Valdgrid, while still providing useful crash reports.

Furthermore, we now have a number of systems that depend on the regtest output format, like the dashboard, regtest survey, and the CI. The bottom line is that `do_regtest` solves a problem that is very specific to us, but not hard. Hence, I don't see an advantage of using an external framework.
